fix: change callout metadata regex to include non-letter characters

HyperEpsilon opened the pull request:
Fix for oversight in #1172. Regarding #1110 

Previous implementation did not allow for spaces or non-letter characters in the callout metadata

HyperEpsilon committed:
fix: change callout metadata regex to include non-letter characters

MasssiveJuice08 commented:
Absolute godsend adding this, thanks so much @HyperEpsilon ! This will make adapting custom callouts from Obsidian for Quartz so much easier and more versatile.

HyperEpsilon committed:
fix: make metadata regex non-greedy

This allows for users to have callouts such as
> [!NOTE|left foo-bar 123] a ]+ title with square brackets [s] a
> Contents

HyperEpsilon commented:
Last edit. Made the regex non-greedy so if there are square brackets in the callout title, the parser doesn't think they are part of the metadata
